ops/terraform/services/pipeline/modules/bfd_pipeline_slis/lambda_src/update_pipeline_slis.py

```diff
@@ -0,0 +1,629 @@
+                print(
+                    f'Retrieving metric data from "{METRICS_NAMESPACE}/{data_first_available_name}"'
+                    f" with dimensions {group_timestamp_dimension}..."
+                )
+                result = backoff_retry(
+                    func=get_data_first_available_for_group,
+                    ignored_exceptions=COMMON_UNRECOVERABLE_EXCEPTIONS + [KeyError],
+                )
+                print(
+                    f'Metric "{METRICS_NAMESPACE}/{data_first_available_name}" with dimensions'
+                    f" {group_timestamp_dimension} retrieved successfully"
+                )
+            except Exception as exc:
+                print(
+                    f"An unrecoverable error occurred when trying to call GetMetricData; err: {exc}"
+                )
+                return
+
+            if [
+                x
+                for x in result
+                if x.label == f"{METRICS_NAMESPACE}/{data_first_available_name}" and x.values
+            ]:
+                print(
+                    f'Metric data exists for "{METRICS_NAMESPACE}/{data_first_available_name}" with'
+                    f" dimensions {group_timestamp_dimension}. Incoming file is part of an ongoing,"
+                    " existing data load, and therefore does not indicate the time of the first"
+                    " data load for its group. Stopping..."
+                )
+                return
+
+            print(
+                "No metric data result was found for metric"
+                f" {METRICS_NAMESPACE}/{data_first_available_name}, this indicates that the"
+                " incoming file is the start of a new data load. Putting data to metric"
+                f' "{METRICS_NAMESPACE}/{data_first_available_name}" with value {utc_timestamp}'
+            )
+
+            def put_data_first_available_metrics():
+                return put_metric_data(
+                    metric_namespace=METRICS_NAMESPACE,
+                    metrics=gen_all_dimensioned_metrics(
+                        metric_name=data_first_available_name,
+                        dimensions=[group_timestamp_dimension],
+                        timestamp=event_timestamp,
+                        value=utc_timestamp,
+                        unit="Seconds",
+                    ),
+                )
+
+            try:
+                print(
+                    f'Putting time metric data to "{METRICS_NAMESPACE}/{data_first_available_name}"'
+                    f" with value {utc_timestamp}..."
+                )
+                backoff_retry(
+                    func=put_data_first_available_metrics,
+                    ignored_exceptions=COMMON_UNRECOVERABLE_EXCEPTIONS,
+                )
+                print("Data put successfully")
+            except Exception as exc:
+                print(
+                    f"An unrecoverable error occurred when trying to call PutMetricData; err: {exc}"
+                )
```

Comment:
Another case where we have an unrecoverable error but don't return; is it that we can continue on the puts failing but not the gets?

Reply:
Same deal as above: looks like I forgot to include a `return` here. Thanks for catching this!